system/lib/pthread/library_pthread.c

```diff
@@ -751,32 +768,63 @@ int _emscripten_call_on_thread(int forceAsync, pthread_t targetThread, EM_FUNC_S
   // (we also never will in that case).
   if (!q)
     return 0;
-  q->functionEnum = sig;
-  q->functionPtr = func_ptr;
-  q->satelliteData = satellite;
 
+  // `q` will not be used after it is called, so let the call clean it up.
+  q->calleeDelete = 1;
+  return _emscripten_do_dispatch_to_thread(target_thread, q);
+}
+
+int emscripten_dispatch_to_thread_(pthread_t target_thread,
+                                   EM_FUNC_SIGNATURE sig,
+                                   void* func_ptr,
+                                   void* satellite,
+                                   ...) {
   va_list args;
   va_start(args, satellite);
-  init_em_queued_call_args(q, sig, args);
+  int ret = emscripten_dispatch_to_thread_args(
+    target_thread, sig, func_ptr, satellite, args);
```

Comment:
Is this what clang-format wants?

Reply:
Yes, but apparently I missed a different formatting error. Just pushed a fix.